File: hackerrank/easy/0023-viral-advertising/swift/solution.swift
// solution.swift

func viralAdvertising(n: Int) -> Int {
    var cumulative = 0
    var shared = 5
    for _ in 1...n {
        let liked = shared / 2
        cumulative += liked
        shared = liked * 3
    }
    return cumulative
}

// --- demo run ---
print(viralAdvertising(n: 3)) // 9
print(viralAdvertising(n: 5)) // 24

File: hackerrank/easy/0023-viral-advertising/js/solution.mjs
// solution.mjs

export function viralAdvertising(n) {
  let cumulative = 0;
  let shared = 5;
  for (let day = 1; day <= n; day++) {
    const liked = Math.floor(shared / 2);
    cumulative += liked;
    shared = liked * 3;
  }
  return cumulative;
}

// --- demo run ---
console.log(viralAdvertising(3)); // 9
console.log(viralAdvertising(5)); // 24